### ISSUE TITLE
Compile nginx for SSL 

### ISSUE THREAD

**Issue by robclay** [Wonderfall/dockerfiles#179]
I just pulled the latest nextcloud:11.0
I tried editing the nginx conf file, to point to my purchased SSL certificate.
I get the following error:
`nginx: [emerg] the "ssl" parameter requires ngx_http_ssl_module in /nginx/sites-enabled/nginx.conf:2`
How do I install that module?

Thanks for any assistance!

**Comment by Wonderfall** [comment]
Hi,
The nextcloud container was not designed to be directly exposed to the Internet. You'll need a reverse proxy to do that.

**Comment by Steiniche** [comment]
@robclay you could use something like jwilder/nginx-proxy to do the reverse proxy.
Furthermore, you could have used the jrcs/letsencrypt-nginx-proxy-companion to get free letsencrypt certificates.

@Wonderfall I believe this issue should be closed?

**Comment by robclay** [comment]
@Steiniche does jrcs/letsencrypt-nginx-proxy-companion work even when you do not have access to port 80?

**Comment by Steiniche** [comment]
@robclay no you will need access to port 80 for letsencrypt to verify.

**Comment by robclay** [comment]
I'm hosting from home... So the purchased cert works as long as I https... (Port 443 is allowed)  I type this in case anyone else is thinking about doing so, in the same situation.  I think Wonderfall's 11.0 or 11.01 build allowed me to go into the container and edit nginx.conf at /etc/nginx and turn ssl on.  The nginx conf file has moved in the 11.03 build.  /nginx/sites-_something_  .  But I understand that Wonderfalls intention is not to host directly...

Thank you both for your time!

**Comment by robclay** [comment]
So I am doing the following:
docker pull wonderfall/boring-nginx
docker run --link nextcloud:nextcloud -v /shares/ncconfig/RapidSSL:/certs -d wonderfall/boring-nginx
docker exec -ti CONT_ID ngxproxy

The scrip asks for the webroot?  do I accept the default?
And then I get a 
WARNING: CONT_ID seems to be unavailable. It may not work!
HINT: Did you correctly link the container?

Is my link command not correct in the run command above?

Thanks for any assistance!

**Comment by Wonderfall** [comment]
It is, if you don't rename your containers. Use `--name` if you don't want to use docker-compose (though it's highly recommended).

**Comment by robclay** [comment]
I think I am getting closer...
After running the command above, I ran
docker exec -ti CONT_ID ngxproxy
Name: nextcloud
Domain: www.DOMAIN.com
Webroot (default is /):
Container: nextcloud
Port (default is 80): 8888
HTTPS [y/n]: n
Max body size in MB (integer/null): null

This was just to see if it would talk to the docker container...  i.e. I am not even trying to fight the SSL stuff.

In my docker container I added into my trusted domains:
5 => 'IP.IP.IP.IP:8000',

Since a docker ps shows the boring-nginx at 8000/tcp.  

And then when I use my web browser to go to IP.IP.IP.IP:8000, it states firefox can’t establish a connection to the server at IP.IP.IP.IP:8000.  (The IP is an internal IP)

Thanks for any and all help (And patience!)
